blockbuster/abci.go

```diff
@@ -56,15 +91,73 @@ func (h *ProposalHandler) PrepareProposalHandler() sdk.PrepareProposalHandler {
 	}
 }
 
+// ProcessProposalHandler processes the proposal by verifying all transactions in the proposal
+// according to each lane's verification logic. We verify proposals in a greedy fashion.
+// If a lane's portion of the proposal is invalid, we reject the proposal. After a lane's portion
+// of the proposal is verified, we pass the remaining transactions to the next lane in the chain.
 func (h *ProposalHandler) ProcessProposalHandler() sdk.ProcessProposalHandler {
 	return func(ctx sdk.Context, req abci.RequestProcessProposal) abci.ResponseProcessProposal {
-		for _, l := range h.mempool.registry {
-			if err := l.ProcessLane(ctx, req.Txs); err != nil {
-				h.logger.Error("failed to process lane", "lane", l.Name(), "err", err)
-				return abci.ResponseProcessProposal{Status: abci.ResponseProcessProposal_REJECT}
-			}
+		if _, err := h.processLanesHandler(ctx, req.Txs); err != nil {
+			h.logger.Error("failed to process lanes", "err", err)
+			return abci.ResponseProcessProposal{Status: abci.ResponseProcessProposal_REJECT}
 		}
 
 		return abci.ResponseProcessProposal{Status: abci.ResponseProcessProposal_ACCEPT}
 	}
 }
+
+// Terminator Lane will get added to the chain to simplify chaining code so that we
+// don't need to check if next == nil further up the chain
+type Terminator struct{}
```

Comment:
```suggestion
type TerminatorLane struct{}
```

Feels weird to define this in `blockerbuster` and not `blockbuster/lanes/terminator`.